common/src/main/java/xyz/wagyourtail/jsmacros/client/api/helpers/EntityHelper.java

```diff
@@ -315,11 +581,210 @@ public LivingEntityHelper<?> asLiving() {
         return (LivingEntityHelper<?>) this;
     }
 
+    /**
+     * @return this helper as an animal entity helper (mainly for typescript).
+     *
+     * @since 1.8.4
+     */
+    public LivingEntityHelper<?> asAnimal() {
+        return (AnimalEntityHelper<?>) this;
+    }
+
+    /**
+     * @return a specialized helper for this entity if it exists and self otherwise.
+     *
+     * @since 1.8.4
+     */
+    public EntityHelper<?> getSpecialized() {
```

Comment:
I will move it to `Entity.create`. I'm not sure about the HashMap because the instanceOf also handles modded mobs, which would be quite complicated to achieve with a map. The speed loss is probably neglectable even with instanceOf.

Reply:
It should be possible to use the `EntityType` registry to get all loaded mobs and their respective classes.

Reply:
well, if its not in the map, we'd go back through with an `isAssignableTo`, the problem would be ordering